### PR TITLE
Remove unused feature flags for upgrades/*

### LINKED ISSUE #1301: Domains: Remove feature flags
We should remove [feature flags](https://github.com/Automattic/wp-calypso/blob/master/config/development.json#L86) for the domains management section - for all environments (similarly to what we did for the purchases management section in https://github.com/Automattic/wp-calypso/issues/1028):

```
upgrades/domain-management/contacts-privacy
upgrades/domain-management/email
upgrades/domain-management/list
upgrades/domain-management/name-servers
upgrades/domain-management/site-redirect
upgrades/domain-management/transfer
```

### PR DESCRIPTION
**Pull request by umurkontaci**
Fixes #1301.

This removes feature flags that are enabled for all environments under upgrades/*

No behaviour changes.

Testing:
- Ensure URLs load properly
- Ensure there are no remaining references to the removed flags anywhere in the code.

/cc: @stephanethomas @klimeryk

### CODE REVIEW

File: client/my-sites/upgrades/index.js [hunk 1]
```diff
@@ -222,38 +212,19 @@ module.exports = function() {
...
+		controller.siteSelection,
+		controller.sites
+	);
 
-		page( '/domains/:domain',
-			adTracking.retarget,
-			controller.siteSelection,
-			controller.navigation,
-			controller.jetPackWarning,
-			upgradesController.domainSearchIndex
```
Comment (klimeryk) [inline client/my-sites/upgrades/index.js]: It looks like with this change `upgradesController.domainSearchIndex` becomes unused anywhere in our code and can be removed :raised_hands:

File: client/my-sites/current-site/index.jsx [hunk 2]
```diff
@@ -95,11 +95,7 @@ module.exports = React.createClass( {
 			return null;
 		}
 		const { hostname } = url.parse( site.URL );
-		let href = 'https://wordpress.com/my-domains';
-
-		if ( config.isEnabled( 'upgrades/domain-management/list' ) ) {
-			href = paths.domainManagementList( site.domain );
-		}
+		let href = paths.domainManagementList( site.domain );
```
Comment (klimeryk) [inline client/my-sites/current-site/index.jsx]: With this change, `href` can also be a `const` :)

**Comment by klimeryk** [comment]
Two small issues, but other than that everything looks good - nice improvement, less conditionals/nesting.

**Comment by klimeryk** [comment]
Interesting - although you've removed the `upgrades/domain-management/transfer` flag in #2883, it seems there are still some leftovers of it in [`desktop-mac-app-store.json`](https://github.com/Automattic/wp-calypso/blob/update/1301-remove-unused-feature-flags/config/desktop-mac-app-store.json#L82), [`desktop.json`](https://github.com/Automattic/wp-calypso/blob/update/1301-remove-unused-feature-flags/config/desktop.json#L82) and [`development.json`](https://github.com/Automattic/wp-calypso/blob/update/1301-remove-unused-feature-flags/config/development.json#L119). It's not actually referenced in code, it can be safely removed.

File: client/my-sites/current-site/index.jsx (hunk 2 above)
Comment (stephanethomas) [inline client/my-sites/current-site/index.jsx]: I think we could even remove this variable and inline this code inside `<NoticeAction />` below.

**Comment by stephanethomas** [comment]
Nice changes. And kudos for splitting them into separate commits.

**Comment by umurkontaci** [comment]
Thanks for the feedback! I've addressed them all.

**Comment by klimeryk** [comment]
LGTM!

**Pull request merged**